### PR TITLE
Fix/issue 1196

### LINKED ISSUE #1196: Backport openfe fixes for context deletion
From https://github.com/OpenFreeEnergy/openfe/pull/354 there are some fixes that may be of interest for us for handling context creatiopn/deletion. If we fix https://github.com/choderalab/openmmtools/pull/690 this may not be needed.

### PR DESCRIPTION
**Pull request by mikemhenry**
## Description
Back ports some fixes from the OpenFE team.

## Motivation and context

Resolves #1196 

## Change log

```

```
Explicitly deleted some sampler and energy contexts in `setup_relative_calculation.py`

### CODE REVIEW

File: perses/app/setup_relative_calculation.py [hunk 1]
```diff
@@ -753,7 +753,7 @@ def run_setup(setup_options, serialize_systems=True, build_samplers=True):
 def run(yaml_filename=None, override_string=None):
     cli_tool_name = sys.argv[0].split(os.sep)[-1]
     if cli_tool_name == "perses-relative":
-        warnings.warn("perses-relative will be removed in 0.11, see https://github.com/choderalab/perses/tree/main/examples/new-cli for new CLI tool", FutureWarning)
+        warningg.warn("perses-relative will be removed in 0.11, see https://github.com/choderalab/perses/tree/main/examples/new-cli for new CLI tool", FutureWarning)
```
Comment (ijpulidos) [inline perses/app/setup_relative_calculation.py:756]: Typo? `warningg`

**Review by ijpulidos — CHANGES_REQUESTED** [review]
Seems good, thanks for this! Just a minor change, probably a typo or some accident :)

File: perses/app/setup_relative_calculation.py (hunk 1 above)
Comment (mikemhenry) [reply perses/app/setup_relative_calculation.py:756]: lmao, yes, I was vim-ing

Comment (mikemhenry) [inline perses/app/setup_relative_calculation.py:756]:
```suggestion
        warning.warn("perses-relative will be removed in 0.11, see https://github.com/choderalab/perses/tree/main/examples/new-cli for new CLI tool", FutureWarning)
```

**Review by ijpulidos — APPROVED** [review]
LGTM!

**Pull request merged**